### PR TITLE
enabling hsts middleware (search-ui)

### PR DESCRIPTION
**Pull request by Anxhela21**
**Related Issue:**  open-cluster-management/backlog#13465

### Description of changes
- Enabling HSTS on the search-ui response headers

### CODE REVIEW

File: backend/src/server.ts [hunk 1]
```diff
@@ -18,9 +18,27 @@ import { join } from 'path'
...
 
+// const server = fastify()
+// const helmet = require('fastify-helmet')
+// fastify.use(require('hsts')())
+// server.register(helmet, { hsts: true })
+// const fastify = require('fastify')
+
+// fastify.use(require('hsts')())
+
+const fastify = require('fastify')()
```
Comment (jlpadilla) [inline backend/src/server.ts:33]: - fastify is imported in line 6 as `Fastify` (uppercase). You don't need this import.
- The `fastify` (lowercase) instance is created below around line 66-70.  So you need to move `fastify.register(...)` after line 70.

File: backend/src/server.ts [hunk 2]
```diff
@@ -18,9 +18,27 @@ import { join } from 'path'
...
+const helmet = require('fastify-helmet')
```
Comment (jlpadilla) [inline backend/src/server.ts:34]: The preferred way to import this library is
```
import fastifyHelmet from 'fastify-helmet'
```
it does the same as require(), but typescript gets unhappy with require.

**Comment by Anxhela21** [comment]
image

**Comment by Anxhela21** [comment]
Each directive has nonce applied as an extra layer of security. I am not seeing any directive errors however coming across a certificate error:

image

When validating csp and hsts security headers for search we can see they are present (see highlights below):

image

**Comment by Anxhela21** [comment]
Certificate issue is independent of security header changes.

**Comment by Anxhela21** [comment]
After some more investigation, HSTS is already enabled for Search UI.

**Pull request closed without merging**